### PR TITLE
TEST: PL_atom_wchars print

### PR DESCRIPTION
**Pull request by kamahen**
For debugging (and regression testing) https://github.com/SWI-Prolog/packages-pcre/issues/20

### CODE REVIEW

File: test_ffi.pl [hunk 1]
```diff
@@ -91,4 +91,32 @@
 test(range_ffialloc5, X == 1) :- % Will produce warning if non-deterministic
     range_ffialloc(1, 2, X).
 
+
+% The following "wchar" tests are regression tests related
+% to https://github.com/SWI-Prolog/packages-pcre/issues/20
+
+test(wchar_1, all(Result == ["//0", "/ /1", "/abC/3", "/Hello World!/12", "/хелло/5", "/хелло 蛇/7", "/網目錦へび [àmímé níshíkíhéꜜbì]/26"])) :-
```
Comment (JanWielemaker) [inline test_ffi.pl]: I'm a bit afraid having UTF-8 encoded source files in the tree.   Unfortunately I think we are still in the position that transferring such files may easily reencode them.    The file surely requires `:- encoding(utf8).` to work cross platform.    All the other tests use `\uXXXX` or generate unicode characters in a range using e.g. numlist/3.

Comment (kamahen) [reply test_ffi.pl]: Good point about setting the encoding (sorry, I forgot about that).

`test_pcre.pl` has UTF8 in it ... has that caused a problem?

BTW, I tried making this into a separate test but it didn't work -- is it possible to have more than one `swipl_plugin(TESTONLY)` in a `CMakeLists.txt`?

Comment (JanWielemaker) [reply test_ffi.pl]: > `test_pcre.pl` has UTF8 in it ... has that caused a problem?

Not AFAIK.   So maybe we can try and switch to `\uXXXX` if there are problems that are hard to bypass some other way.   Moving to UTF-8 seems similar to IPv6 :cry: 

> separate test

Of course you can define multiple foreign modules using multiple swipl_plugin().   You can also have multiple test_*.pl files by adding more names to test_libs().   Any name you add there expects a file test_name.pl.   In this case though, I would use multiple plunit test blocks (:- begin_tests(...)) and run all of them from test_ffi/0.

I think you want to do that, so I'll hold merging this PR.

**Comment by kamahen** [comment]
I have made the begin_tests change you suggested and also added a "coding" line at the beginning (for Emacs; I think that there's a way to make this work with vim, but I don't know how and Google didn't help my find a way).

**Pull request merged**